src/tree.ml

```diff
@@ -686,18 +706,25 @@ let rec compile : CM.t -> tree = fun ({clauses ; positions ; slot} as pats) ->
       in
       (* Default child *)
       let default =
+        let compile = compile vars_id in
         let (positions, clauses) = CM.default swap positions updated in
         let ncm = CM.{clauses ; slot ; positions} in
         if CM.is_empty ncm then None else Some(compile ncm)
       in
       (* Abstraction specialisation*)
       let abstraction =
-        if List.for_all (fun (x, _) -> x <> TC.Abst) cons then None else
-        let var = Bindlib.new_var mkfree "tr" in
+        let is_abst = function Abst(_) -> true | _ -> false in
+        if List.for_all (fun x -> not (is_abst x)) column then None else
+        let var = Bindlib.new_var mkfree var_prefix in
```

Comment:
Inline var_prefix here.

Reply:
Isn't it better to have a constant rather that several times the "var" string written in the code?

Reply:
But var_prefix is used only here in tree.ml... Using a constant usable outside suggests that the choice of this prefix is important which is not the case. In fact you could take any string here, including "".